components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/impl/UserApiServiceImpl.java

```diff
@@ -48,7 +48,7 @@ public Response privilegedUserListMediaInformation(String type, String id) {
     public Response privilegedUserUploadMedia(String type, List<InputStream> filesInputStream,
                                               List<Attachment> filesDetail, PrivilegedUserMetadata metadata) {
 
-        mediaService.validateMediaType(type, filesDetail.get(0).getContentType());
+        mediaService.validateAllowedMediaTypesForUploading(type, filesDetail.get(0).getContentType());
```

Comment:
shall we change the method name to, validateFileUploadMediaTypes or something similar?